server/src/test/java/org/apache/druid/server/coordinator/duty/CompactSegmentsTest.java

```diff
@@ -270,12 +275,246 @@ public String get()
     assertLastSegmentNotCompacted(compactSegments);
   }
 
+  @Test
+  public void testMakeStats()
```

Comment:
Hmm this test looks similar to `testRun()`. Can they be merged by moving the snapshot verification to `assertCompactSegments()` (or merging `assertCompactSegmentStatistics` and `assertCompactSegments`)?

Reply:
I feel that each test is already doing a lot. Combining them will make the tests hard to understand as it will be testing and verifying many many many things. They are also not duplicated code as they are verifying different things so I think it is fine to separate them.

Reply:
For example, if you only care about the actual compaction (i.e.the spec in the compaction is not as expected, etc) then you can focus on testRun() and ignore all the stuff in assertCompactSegmentStatistics (along with all the complicated calculation of each stats in each compaction call). 
On the other hand if you need to debug incorrect calculation in bytesCompacted (but the actual compaction tasks are fine) then you can focus on testMakeStats()